resolwe/storage/manager.py

```diff
@@ -18,251 +18,307 @@
+                created__lt=now()  # type: ignore
+                - models.Case(
+                    *whens,
+                    default=timedelta(days=default_delay),
+                    output_field=models.DurationField(),
+                ),
+            )
+            .annotate(
+                all_locations_count=models.Count(
+                    "storage_locations",
+                    filter=models.Q(
+                        storage_locations__status=StorageLocation.STATUS_DONE
+                    ),
+                    distinct=True,
+                )
+            )
+            .annotate(
+                open_logs=models.Exists(
+                    AccessLog.objects.filter(
+                        storage_location__file_storage=models.OuterRef("id"),
+                        storage_location__connector_name=connector_name,
+                        finished__isnull=True,
+                    )
+                )
+            )
+            .annotate(
+                transfered_lately=models.Exists(
+                    AccessLog.objects.filter(
+                        storage_location__file_storage=models.OuterRef("id"),
+                        storage_location__connector_name=connector_name,
+                        started__gt=now()  # type: ignore
+                        - models.Case(
+                            *whens,
+                            default=timedelta(days=default_delay),
+                            output_field=models.DurationField(),
+                        ),
+                    )
+                )
+            )
+            .annotate(
+                accessed_lately=models.Exists(
+                    StorageLocation.objects.filter(
+                        file_storage=models.OuterRef("id"),
+                        last_update__gt=now()  # type: ignore
+                        - models.Case(
+                            *whens,
+                            default=timedelta(days=default_delay),
+                            output_field=models.DurationField(),
+                        ),
+                        connector_name=connector_name,
+                    )
+                )
+            )
+            .annotate(
+                default_connector=models.functions.Coalesce(
+                    models.Subquery(
+                        StorageLocation.objects.filter(
+                            file_storage=models.OuterRef("id"),
+                            status=StorageLocation.STATUS_DONE,
+                        )
+                        .annotate(
+                            priority=models.Case(
+                                *default_connector_whens,
+                                default=DEFAULT_CONNECTOR_PRIORITY,
+                                output_field=models.IntegerField(),
+                            )
+                        )
+                        .order_by("priority")
+                        .values_list("connector_name", flat=True)[:1]
```

Comment:
Can you use `.first()` instead of `[:1]`?

Reply:
Unfortunately not. I had to refresh my memory but it has something to do with first being eager and not lazy and can not be used in such subquery.